org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/matching/MatchLocator.java

```diff
@@ -1002,6 +1002,11 @@ public MethodBinding getMethodBinding(MethodPattern methodPattern) {
     				if (unit != null) {
     					AbstractMethodDeclaration amd = new ASTNodeFinder(unit).findMethod((IMethod) methodPattern.focus);
     					if (amd != null && amd.binding != null && amd.binding.isValidBinding()) {
+    						// Check if we "forgot" to resolve parameter types, see: https://github.com/eclipse-jdt/eclipse.jdt.core/issues/821
+    						if (amd.binding.parameterNames != null && amd.binding.parameters != null &&
+    								amd.binding.parameterNames.length != amd.binding.parameters.length) {
+    							sourceTypeBinding.resolveTypesFor(amd.binding);
```

Comment:
@gayanper , are you sure we shouldn't keep this? From my debugging, as soon as the code is here the parameter types might be unresolved. I've not tried to construct a case where this is a problem, but maybe there are such cases.

Reply:
If we can find a situation where the closest match doesn't resolve, I think we can have this, But i think for this specific issue the closest match solve the resolution issue right ?

But as you say we might have other scenarios where method bindings are not property resolved, when looking at the implementation of `resolveTypesFor` it actually use `resolveTypesWithSuspendedTempErrorHandlingPolicy` which sounds like tring to resolve ignore any errors. So using this method might be usefull in some scenarios.

If we keep it, I would like we make the comment more clear rather than highlighting it as a fix for this issue since it is not anymore at least from the scenarios we know right ?

What if we have a method which has some issues like incompatible generic usage or some issue with parameters definitions (not access issue like this one), will this code block helps to still resolve the call hierarchy ?

I'm more leaning towards removing this block unless it doesn't solve any scenario related to search. @iloveeclipse  WDYT ?